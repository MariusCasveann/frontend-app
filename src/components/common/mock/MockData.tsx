import { GraphQLError } from 'graphql';
import { QueryAllProducts_allProducts } from './../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryAllDomains_allDomains } from './../../dashboard/atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { LightApplicationsOfProduct_lightApplicationsOfProduct } from './../../applications/graphql/__generated__/LightApplicationsOfProduct';
import { QueryBilling_billingQuery } from '../../dashboard/molecules/common/qraphql/__generated__/QueryBilling';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from '../../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { SelectItemProps } from '../molecules/SearchableSelect/SearchableSelectPresenter';

export const mockGraphQLError: GraphQLError = {
    message: 'FATALITY',
    name: 'error',
    locations: undefined,
    path: undefined,
    nodes: undefined,
    positions: undefined,
    source: undefined,
    originalError: undefined,
    extensions: undefined
};

export const mockSelectedDomain: QueryAllDomains_allDomains = {
    id: 102,
    name: 'Engineering Platform',
    __typename: 'Domain'
};

export const mockSelectedProduct: QueryAllProducts_allProducts = {
    id: 50051,
    logo: null,
    name: 'a',
    __typename: 'LightProductDTO'
};

export const mockedApplications: LightApplicationsOfProduct_lightApplicationsOfProduct[] = [
    {
        id: 50051,
        staticPictureId: null,
        name: 'a',
        service: false,
        __typename: 'LightApplicationDTO'
    }
];

export const mockFunction = jest.fn();

export const mockDataEmpty: QueryBilling_billingQuery[] = [];

// tslint:disable-next-line:no-any
export const createMockedState = (loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const userProductMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const preSelectedProduct: QueryAllProducts_allProducts = {
    __typename: 'LightProductDTO',
    id: 'ALL',
    logo: 'ALL',
    name: 'All Products'
};

export const currentUserMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: preSelectedProduct
        }
    }
});

export const selectedProductMockedState = (
    loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>
): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const buildDataExpectedEmptyResults = {
    labels: [],
    datasets: [
        {
            data: []
        }
    ]
};

export const mockErrorMessage = 'Error message';

export const mockTeamMember: TeamByProductId_teamByProductId_teamMembers = {
    __typename: 'User',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    lastName: 'Test',
    id: '11',
    roles: [
        {
            name: 'WRITE_ALL',
            __typename: 'Role'
        },
        {
            name: 'a-admin',
            __typename: 'Role'
        }
    ]
};

export const mockTeamByProductId: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: 2355,
    productOwner: {
        id: 11,
        firstName: 'TestUser',
        lastName: 'Test',
        __typename: 'User'
    },
    teamMembers: [
        {
            __typename: 'User',
            email: 'test1@mediamarktsaturn.com',
            firstName: 'Test',
            id: '1234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                }
            ]
        },
        {
            __typename: 'User',
            email: 'test2@mediamarktsaturn.com',
            firstName: 'Test2',
            id: '2234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                },
                {
                    name: 'test-admin',
                    __typename: 'Role'
                }
            ]
        }
    ]
};

export const mockCurrentUser: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: [
        {
            concretePermissions: [],
            id: 1300,
            name: 'DP God',
            readOnly: true,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 4651,
            name: 'READ_ALL',
            readOnly: false,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 69002,
            name: 'a-read',
            readOnly: true,
            __typename: 'Role'
        }
    ],
    staticPictureId: null,
    subscribedToNewsletter: true
};

export const mockEmptyTeam: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: null,
    productOwner: null,
    teamMembers: []
};

export const mockCurrenUserRole: QueryCurrentUser_currentUser_roles = {
    concretePermissions: [],
    id: 1300,
    name: 'DP God',
    readOnly: true,
    __typename: 'Role'
};

export const mockForm = {
    getFieldDecorator: () => () => false,
    getFieldError: mockFunction,
    getFieldsError: mockFunction,
    getFieldsValue: mockFunction,
    getFieldValue: mockFunction,
    isFieldValidating: mockFunction,
    setFields: mockFunction,
    setFieldsValue: mockFunction,
    validateFields: mockFunction,
    validateFieldsAndScroll: mockFunction,
    isFieldsTouched: mockFunction,
    isFieldTouched: mockFunction,
    resetFields: mockFunction
};

export const mockHistory = {
    length: 7,
    push: mockFunction,
    action: 'PUSH',
    location: {
        hash: '',
        pathname: '/path',
        search: '',
        state: {}
    },
    replace: mockFunction,
    go: mockFunction,
    goBack: mockFunction,
    goForward: mockFunction,
    block: mockFunction,
    createHref: mockFunction,
    listen: mockFunction
};

export const mockSelectData: SelectItemProps[] = [
    {
        id: '351952',
        label: 'Test test'
    },
    {
        id: '80503',
        label: 'Test1 test1'
    },
    {
        id: '412605',
        label: 'Test2 test2'
    }
];
