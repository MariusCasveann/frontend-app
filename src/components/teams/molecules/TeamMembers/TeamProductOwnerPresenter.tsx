import React from 'react';
import { TeamByProductId_teamByProductId } from './graphql/__generated__/TeamByProductId';
import { Avatar, Icon, Button } from 'antd';
import './Team.css';

interface TeamProductOwnerPresenter {
    teamByProductId: TeamByProductId_teamByProductId;
    setChangeProductOwner: (flag: boolean) => void;
    setAddTeamMembers: (flag: boolean) => void;
}

export default ({ teamByProductId, setChangeProductOwner, setAddTeamMembers }: TeamProductOwnerPresenter) => (
    <div className="product-owner-card">
        <div className="product-owner-label">Product owner: </div>
        <Avatar size="small" icon="user" />
        <span className="product-owner-name">
            {teamByProductId && teamByProductId.productOwner && teamByProductId.productOwner.firstName}{' '}
            {teamByProductId && teamByProductId.productOwner && teamByProductId.productOwner.lastName}
        </span>
        <Icon className="edit-icon" type="edit" onClick={() => setChangeProductOwner(true)} />
        <Button className="add-member-button" type="primary" onClick={() => setAddTeamMembers(true)}>
            <Icon type="user-add" />
            Add member
        </Button>
    </div>
);
