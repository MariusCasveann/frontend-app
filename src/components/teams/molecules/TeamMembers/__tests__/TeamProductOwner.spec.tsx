import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('antd', () => ({ Avatar: 'Avatar', Icon: 'Icon', Button: 'Button' }));
import TeamProductOwnerPresenter from '../TeamProductOwnerPresenter';
import { mockEmptyTeam, mockTeamByProductId } from '../../../../common/mock/MockData';

describe('TeamProductOwnerPresenter', () => {
    describe('when there is value', () => {
        it('renders a Card with teammembers', () => {
            const wrapper = renderer
                .create(<TeamProductOwnerPresenter teamByProductId={mockTeamByProductId} />)
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
    describe('when there is no data', () => {
        it('renders no data', () => {
            const wrapper = renderer.create(<TeamProductOwnerPresenter teamByProductId={mockEmptyTeam} />).toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
