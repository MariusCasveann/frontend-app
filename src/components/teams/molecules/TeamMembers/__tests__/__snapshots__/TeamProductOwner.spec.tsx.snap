// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamProductOwnerPresenter when there is no data renders no data 1`] = `
<div
  className="product-owner-card"
>
  <div
    className="product-owner-label"
  >
    Product owner: 
  </div>
  <Avatar
    icon="user"
    size="small"
  />
  <span
    className="product-owner-name"
  >
     
  </span>
  <Icon
    className="edit-icon"
    onClick={[Function]}
    type="edit"
  />
  <Button
    className="add-member-button"
    onClick={[Function]}
    type="primary"
  >
    <Icon
      type="user-add"
    />
    Add member
  </Button>
</div>
`;

exports[`TeamProductOwnerPresenter when there is value renders a Card with teammembers 1`] = `
<div
  className="product-owner-card"
>
  <div
    className="product-owner-label"
  >
    Product owner: 
  </div>
  <Avatar
    icon="user"
    size="small"
  />
  <span
    className="product-owner-name"
  >
    TestUser
     
    Test
  </span>
  <Icon
    className="edit-icon"
    onClick={[Function]}
    type="edit"
  />
  <Button
    className="add-member-button"
    onClick={[Function]}
    type="primary"
  >
    <Icon
      type="user-add"
    />
    Add member
  </Button>
</div>
`;
